lms/djangoapps/grades/api/views.py

```diff
@@ -128,28 +137,48 @@ def get(self, request, course_id):
         Return:
             A JSON serialized representation of the requesting user's current grade status.
         """
-        username = request.GET.get('username')
-
-        # only the student can access her own grade status info
-        if request.user.username != username:
-            log.info(
-                'User %s tried to access the grade for user %s.',
-                request.user.username,
-                username
-            )
-            return self.make_error_response(
-                status_code=status.HTTP_404_NOT_FOUND,
-                developer_message='The user requested does not match the logged in user.',
-                error_code='user_mismatch'
-            )
 
+        # Use the request user's if none provided.
+
+        if 'username' in request.GET:
+            username = request.GET.get('username')
+        else:
+            username = request.user.username
+
+        # Returns a 404 response if course_id is invalid, or request.user is not enrolled in the course
         course = self._get_course(course_id, request.user, 'load')
         if isinstance(course, Response):
             return course
 
+        elif request.user.username != username:
```

Comment:
Nit: Consider extracting this logic out into a separate helper method, like `_get_effective_user`, for improved code readability.